File: Solutions/PAT/Advanced/1001.cpp
#include <iostream>
#include <iomanip>
using namespace std;
// A+B Format (20)

void prt(int x) {
  int low = x % 1000;
  x /= 1000;
  if (x) {
    prt(x);
    cout << ',' << setw(3) << setfill('0') << low;
  } 
  else cout << low;
}
int main() {
  int a,b;
  cin >> a >> b;
  a += b;
  if (a < 0) { cout << '-'; a = -a; }
  prt(a);
  cout << endl;
  return 0;
}

File: Solutions/PAT/Advanced/1001.rb
#!/usr/bin/env ruby
# A+B Format (20)

def prt(x)
  low = x % 1000
  high = x / 1000
  if high == 0 then
    print low
  else
    prt high
    printf ',%03d', low
  end
end
s = gets.split.map {|e| e.to_i}.reduce :+
print '-' if s < 0
prt s.abs